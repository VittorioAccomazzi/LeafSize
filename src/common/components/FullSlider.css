.resultLabel {
    border: 1px;
    border-color: lightgray;
    border-style: solid;
    text-align: left;
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
    color: lightgray;
    min-width: 6em;
    max-width: 6em;
}

.titleLabel {
    width:4em;
    text-align: right;
}
