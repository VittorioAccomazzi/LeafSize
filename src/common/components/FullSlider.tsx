import { Stack, FormLabel, Button, Slider } from "@mui/material";
import { useEffect, useState } from "react";
import NavigateNextIcon from '@mui/icons-material/NavigateNext';
import NavigateBeforeIcon from '@mui/icons-material/NavigateBefore';
import './FullSlider.css'

interface FullSliderProp {
    label : string,
    values : string [],
    disabled : boolean,
    defaultIndex? : number,
    onChange : (index : number, value : string )=>void 
    inverted? : boolean
}

export default function FullSlider( { label, values, disabled, defaultIndex =0, onChange: onChanged, inverted = false } : FullSliderProp){
    const [imageNum, setImageNum] = useState<number>(0);

    const handleSliderChange = (event: Event, newValue: number | number []) => {
        const index = newValue as number;
        setImageNum(index );
      };

    // invoke callback when the index changes
    useEffect(()=>{onChanged(imageNum,values[imageNum])},[imageNum]); // eslint-disable-line  react-hooks/exhaustive-deps

    // in case the value array changes, reset the index
    useEffect(()=>setImageNum(0),[setImageNum,values]);

    // set default index
    useEffect(()=>{if( defaultIndex > 0 ) setImageNum(defaultIndex) },[defaultIndex,setImageNum])

    return (
        <Stack direction='row' alignItems='center' spacing={2}>
            <FormLabel className="titleLabel">
                {label}
            </FormLabel>
            <Button 
                data-testid="left-click"
                variant="outlined" 
                onClick={()=>setImageNum(imageNum-1)} 
                disabled={imageNum===0||disabled} size='small'
            >
                <NavigateBeforeIcon fontSize="small"/> 
            </Button>
            <Slider 
                min={0} 
                value={imageNum} 
                max={values.length-1} 
                disabled={disabled} 
                onChange={handleSliderChange} 
                marks 
                sx={{width:"1024px", '& .MuiSlider-track': { height:"5px", border:'none'}, '& .MuiSlider-rail': {height:"5px", border:'none'}}}
                track={inverted ? "inverted" : "normal" } 
            />
            <Button 
                data-testid="right-click"
                variant="outlined" 
                onClick={()=>setImageNum(imageNum+1)} 
                disabled={imageNum===values.length-1||disabled} 
                size='small'
            >
                <NavigateNextIcon fontSize="small"/>
            </Button>
            <FormLabel 
                data-testid="result-label"
                className="resultLabel"
                sx={{fontSize:'small', padding:'2px'}}
                
             >
                {values[imageNum]}
            </FormLabel>
    </Stack>
    )
}