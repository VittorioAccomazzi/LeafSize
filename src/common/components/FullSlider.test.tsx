import { fireEvent, render } from '@testing-library/react';
import FullSlider from './FullSlider'


test('Full Slider Render', ()=>{
    const array = ["one", "two", "three"];
    const wrapper = render( 
        <FullSlider
            label='FullSlider Test Label'
            values={array}
            disabled={false}
            onChange={(i,v)=>{}}
        />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.getByText(/one/i)).toBeInTheDocument();
})

test('Full Slider dynamic behaviour', async ()=>{
    const array = ["one", "two", "three"];
    let index =0;
    let value ="";

    const wrapper = render( 
        <FullSlider
            label='FullSlider Test Label'
            values={array}
            disabled={false}
            onChange={(i,v)=>{index=i; value=v;}}
        />
    );

    const leftButton = (await wrapper.findAllByTestId('left-click'))[0]  as HTMLButtonElement;
    const rightButton= (await wrapper.findAllByTestId('right-click'))[0] as HTMLButtonElement;
    const resultLabel= (await wrapper.findAllByTestId('result-label'))[0] as HTMLLabelElement;

    // left button shall be disabled
    expect(rightButton.disabled).toBeFalsy();
    expect(leftButton.disabled).toBeTruthy();

    // right button shall change the label
    fireEvent.click(rightButton)
    expect(wrapper.getByText(/two/i)).toBeInTheDocument();
    expect(index).toBe(1);
    expect(value).toBe(array[index]);
    expect(resultLabel.innerHTML).toBe('two');

    // right button shall change the label again
    fireEvent.click(rightButton)
    expect(wrapper.getByText(/three/i)).toBeInTheDocument();
    expect(index).toBe(2);
    expect(value).toBe(array[index]);

    // right button shall be disabled
    expect(rightButton.disabled).toBeTruthy();
    expect(leftButton.disabled).toBeFalsy();
})

test ('Full Slider shall reset index when values propery chages',async ()=>{
    const arrayOld = ["old one", "old two", "old three", "old four"];
    const arrayNew = ["new one", "new two" ];

    const wrapper = render( 
        <FullSlider
            label='FullSlider Test Label'
            values={arrayOld}
            disabled={false}
            onChange={(i,v)=>{}}
        />
    );

    const leftButton = (await wrapper.findAllByTestId('left-click'))[0]  as HTMLButtonElement;
    const rightButton= (await wrapper.findAllByTestId('right-click'))[0] as HTMLButtonElement;
    const resultLabel= (await wrapper.findAllByTestId('result-label'))[0] as HTMLLabelElement;

    expect(leftButton.disabled).toBeTruthy();

    fireEvent.click(rightButton); // two
    fireEvent.click(rightButton); // three
    fireEvent.click(rightButton); // four

    expect(rightButton.disabled).toBeTruthy();
    expect(resultLabel.innerHTML).toBe(arrayOld[3]);

    wrapper.rerender(
        <FullSlider
            label='FullSlider Test Label'
            values={arrayNew}
            disabled={false}
            onChange={(i,v)=>{}}
        />
    );

    expect(leftButton.disabled).toBeTruthy();
    expect(rightButton.disabled).toBeFalsy();
    expect(resultLabel.innerHTML).toBe(arrayNew[0]);

})

test ('Full Slider shall set the default index',async ()=>{
    const array = ["one", "two", "three", "four"];
    let index =0;
    let value ="";

    const wrapper = render( 
        <FullSlider
            label='FullSlider Test Label'
            values={array}
            disabled={false}
            defaultIndex={2}
            onChange={(i,v)=>{index=i; value=v;}}
        />
    );

    const leftButton = (await wrapper.findAllByTestId('left-click'))[0]  as HTMLButtonElement;
    const rightButton= (await wrapper.findAllByTestId('right-click'))[0] as HTMLButtonElement;
    const resultLabel= (await wrapper.findAllByTestId('result-label'))[0] as HTMLLabelElement;

    expect(leftButton.disabled).toBeFalsy();
    expect(rightButton.disabled).toBeFalsy();
    expect(resultLabel.innerHTML).toBe(array[2]);
    expect(index).toBe(2);
    expect(value).toBe(array[2]);
})

test('inverted Slider', async ()=>{
    const array = ["one", "two", "three", "four"];
    let index =0;
    let value ="";

    const wrapper = render( 
        <FullSlider
            label='FullSlider Test Label'
            values={array}
            disabled={false}
            defaultIndex={2}
            inverted={true}
            onChange={(i,v)=>{index=i; value=v;}}
        />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.getByText(/three/i)).toBeInTheDocument();
})