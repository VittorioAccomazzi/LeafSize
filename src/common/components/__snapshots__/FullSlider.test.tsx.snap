// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full Slider Render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1yo1jo6-MuiStack-root"
      >
        <label
          class="MuiFormLabel-root MuiFormLabel-colorPrimary titleLabel css-u4tvz2-MuiFormLabel-root"
        >
          FullSlider Test Label
        </label>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root Mui-disabled  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
          data-testid="left-click"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="NavigateBeforeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
        <span
          class="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root css-19rh4qf-MuiSlider-root"
        >
          <span
            class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
          />
          <span
            class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
            style="left: 0%; width: 0%;"
          />
          <span
            class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
            data-index="0"
            style="left: 0%;"
          />
          <span
            class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
            data-index="1"
            style="left: 50%;"
          />
          <span
            class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
            data-index="2"
            style="left: 100%;"
          />
          <span
            class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
            data-index="0"
            style="left: 0%;"
          >
            <input
              aria-orientation="horizontal"
              aria-valuemax="2"
              aria-valuemin="0"
              aria-valuenow="0"
              data-index="0"
              max="2"
              min="0"
              step="1"
              style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
              type="range"
              value="0"
            />
          </span>
        </span>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
          data-testid="right-click"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <label
          class="MuiFormLabel-root MuiFormLabel-colorPrimary resultLabel css-gdirxv-MuiFormLabel-root"
          data-testid="result-label"
        >
          one
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1yo1jo6-MuiStack-root"
    >
      <label
        class="MuiFormLabel-root MuiFormLabel-colorPrimary titleLabel css-u4tvz2-MuiFormLabel-root"
      >
        FullSlider Test Label
      </label>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root Mui-disabled  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
        data-testid="left-click"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="NavigateBeforeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </svg>
      </button>
      <span
        class="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root css-19rh4qf-MuiSlider-root"
      >
        <span
          class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
        />
        <span
          class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
          style="left: 0%; width: 0%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="0"
          style="left: 0%;"
        />
        <span
          class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
          data-index="1"
          style="left: 50%;"
        />
        <span
          class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
          data-index="2"
          style="left: 100%;"
        />
        <span
          class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
          data-index="0"
          style="left: 0%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="2"
            aria-valuemin="0"
            aria-valuenow="0"
            data-index="0"
            max="2"
            min="0"
            step="1"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="0"
          />
        </span>
      </span>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
        data-testid="right-click"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="NavigateNextIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <label
        class="MuiFormLabel-root MuiFormLabel-colorPrimary resultLabel css-gdirxv-MuiFormLabel-root"
        data-testid="result-label"
      >
        one
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`inverted Slider 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1yo1jo6-MuiStack-root"
      >
        <label
          class="MuiFormLabel-root MuiFormLabel-colorPrimary titleLabel css-u4tvz2-MuiFormLabel-root"
        >
          FullSlider Test Label
        </label>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
          data-testid="left-click"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="NavigateBeforeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <span
          class="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root MuiSlider-trackInverted css-19rh4qf-MuiSlider-root"
        >
          <span
            class="MuiSlider-rail css-1xst22u-MuiSlider-rail"
          />
          <span
            class="MuiSlider-track css-1h4p1o1-MuiSlider-track"
            style="left: 0%; width: 66.66666666666667%;"
          />
          <span
            class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
            data-index="0"
            style="left: 0%;"
          />
          <span
            class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
            data-index="1"
            style="left: 33.333333333333336%;"
          />
          <span
            class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
            data-index="2"
            style="left: 66.66666666666667%;"
          />
          <span
            class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
            data-index="3"
            style="left: 100%;"
          />
          <span
            class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
            data-index="0"
            style="left: 66.66666666666667%;"
          >
            <input
              aria-orientation="horizontal"
              aria-valuemax="3"
              aria-valuemin="0"
              aria-valuenow="2"
              data-index="0"
              max="3"
              min="0"
              step="1"
              style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
              type="range"
              value="2"
            />
          </span>
        </span>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
          data-testid="right-click"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <label
          class="MuiFormLabel-root MuiFormLabel-colorPrimary resultLabel css-gdirxv-MuiFormLabel-root"
          data-testid="result-label"
        >
          three
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1yo1jo6-MuiStack-root"
    >
      <label
        class="MuiFormLabel-root MuiFormLabel-colorPrimary titleLabel css-u4tvz2-MuiFormLabel-root"
      >
        FullSlider Test Label
      </label>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
        data-testid="left-click"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="NavigateBeforeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <span
        class="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root MuiSlider-trackInverted css-19rh4qf-MuiSlider-root"
      >
        <span
          class="MuiSlider-rail css-1xst22u-MuiSlider-rail"
        />
        <span
          class="MuiSlider-track css-1h4p1o1-MuiSlider-track"
          style="left: 0%; width: 66.66666666666667%;"
        />
        <span
          class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
          data-index="0"
          style="left: 0%;"
        />
        <span
          class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
          data-index="1"
          style="left: 33.333333333333336%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="2"
          style="left: 66.66666666666667%;"
        />
        <span
          class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
          data-index="3"
          style="left: 100%;"
        />
        <span
          class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
          data-index="0"
          style="left: 66.66666666666667%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="3"
            aria-valuemin="0"
            aria-valuenow="2"
            data-index="0"
            max="3"
            min="0"
            step="1"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="2"
          />
        </span>
      </span>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
        data-testid="right-click"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="NavigateNextIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <label
        class="MuiFormLabel-root MuiFormLabel-colorPrimary resultLabel css-gdirxv-MuiFormLabel-root"
        data-testid="result-label"
      >
        three
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
